x/wasm/types/authz_test.go

```diff
@@ -744,3 +745,184 @@ func mustGrant(contract sdk.AccAddress, limit ContractAuthzLimitX, filter Contra
 	}
 	return *g
 }
+
+func TestValidateCodeGrant(t *testing.T) {
+	specs := map[string]struct {
+		codeHash              []byte
+		instantiatePermission AccessConfig
+		expErr                bool
+	}{
+		"all good": {
+			codeHash:              []byte("ABC"),
+			instantiatePermission: AllowEverybody,
+		},
+		"empty code hash": {
+			codeHash:              []byte{},
+			instantiatePermission: AllowEverybody,
+			expErr:                true,
+		},
+		"nil code hash": {
+			codeHash:              nil,
+			instantiatePermission: AllowEverybody,
+			expErr:                true,
+		},
+		"invalid permission": {
+			codeHash:              []byte("ABC"),
+			instantiatePermission: AccessConfig{Permission: AccessTypeUnspecified},
+			expErr:                true,
+		},
+	}
+	for name, spec := range specs {
+		t.Run(name, func(t *testing.T) {
+			grant, err := NewCodeGrant(spec.codeHash, spec.instantiatePermission)
+			require.NoError(t, err)
+
+			gotErr := grant.ValidateBasic()
+			if spec.expErr {
+				require.Error(t, gotErr)
+				return
+			}
+			require.NoError(t, gotErr)
+		})
+	}
+}
+
+func TestValidateStoreCodeAuthorization(t *testing.T) {
+	validGrant, err := NewCodeGrant([]byte("ABC"), AllowEverybody)
+	require.NoError(t, err)
+	invalidGrant, err := NewCodeGrant(nil, AllowEverybody)
+	require.NoError(t, err)
+
+	specs := map[string]struct {
+		setup  func(t *testing.T) validatable
+		expErr bool
+	}{
+		"all good": {
+			setup: func(t *testing.T) validatable {
+				t.Helper()
```

Comment:
I found this `t.Helper()` not very helpful in table tests therefore I [removed](https://github.com/CosmWasm/wasmd/commit/086c052c1820f9330eb9bef9d785e9963312434c) the linter. Let's keep `setup` functions to the minimum